sdk/src/azure/core/az_json_token.c

```diff
@@ -325,6 +325,52 @@ AZ_NODISCARD static az_result _az_json_token_get_string_helper(
   return AZ_OK;
 }
 
+/**
+ * @brief Gets the JSON token's string after unescaping it, if required.
+ */
+AZ_NODISCARD az_result az_json_string_unescape(
+    az_span json_string,
+    char* destination,
+    int32_t destination_max_size,
+    int32_t* out_string_length)
+{
+  _az_PRECONDITION_VALID_SPAN(json_string, 1, false);
+  _az_PRECONDITION_NOT_NULL(destination);
+  _az_PRECONDITION(destination_max_size > 0);
```

Comment:
We have pretty decent test coverage for the current implementation of this method.

That said, we can improve it slightly by adding test coverage for the precondition validation.

https://dev.azure.com/azure-sdk/public/_build/results?buildId=1743193&view=codecoverage-tab
![image](https://user-images.githubusercontent.com/6527137/181652383-29363db9-17f0-420d-be3f-93a005524a93.png)

Reply:
All nice and dandy , how do you catch exceptions in C ?